sources/functions/qbittorrent

```diff
@@ -41,48 +36,110 @@ function whiptail_qbittorrent() {
     fi
 }
 
+function qbittorrent_version_info() {
+    case $qbittorrent in
+        4.1.*)
+            libtorrent_branch=RC_1_1
+            libtorrent_minimum_version=1.1.0
+            stdcver="c++14"
+            cryptography_type=built-in
+            ;;
+        4.2.*)
+            libtorrent_branch=RC_1_2
+            libtorrent_minimum_version=1.2.0
+            stdcver="c++14"
+            cryptography_type=built-in
+            ;;
+        4.3.*)
+            libtorrent_branch=RC_1_2
+            libtorrent_minimum_version=1.2.11
+            stdcver="c++17"
+            cryptography_type=built-in
+            ;;
+        4.4.*)
+            libtorrent_branch=RC_2_0
+            libtorrent_minimum_version=2.0
+            stdcver="c++17"
+            cryptography_type=built-in
+            ;;
+        *)
+            echo_error "qBittorrent version should be between 4.1.*.* and 4.3.*.* -- Setup will not proceed."
+            exit 1
+            ;;
+    esac
+}
+
+function skip_libtorrent_qbittorrent() {
+    if [[ -f /usr/local/lib/libtorrent.a ]]; then
+        libtorrent_current_version=$(strings /usr/local/lib/libtorrent.a | grep libtorrent/ | cut -d/ -f2)
+        if dpkg --compare-versions ${libtorrent_current_version} ge ${libtorrent_minimum_version}; then
+            if dpkg -s libtorrent-rasterbar | grep -q ${stdcver} && whiptail_skip_libtorrent_rasterbar; then
+                return
+            fi
+        fi
+    fi
+    false
+    return
+}
+
+function build_libtorrent_qbittorrent() {
+    booststrap
+    echo "using gcc : : : <cflags>\"-march=native -std=${stdcver}\" <cxxflags>\"-march=native -std=${stdcver}\" ;" > /root/user-config.jam
+
+    cd /tmp
+    git clone -b ${libtorrent_branch} https://github.com/arvidn/libtorrent >> ${log} 2>&1
```

Comment:
```suggestion
    git clone -b ${libtorrent_branch} https://github.com/arvidn/libtorrent /tmp/libtorrent >> ${log} 2>&1 || {
        echo_error "Failed to clone from github"
        exit 1
    }
```

Reply:
You have suggested changes to outdated code

Reply:
Yikes, my bad!